dask/dataframe/io/parquet/fastparquet.py

```diff
@@ -1179,11 +1179,13 @@ def initialize_write(
                     else None
                 )
                 divisions = division_info["divisions"]
-                if old_end is None or divisions[0] <= old_end:
+                if old_end is not None and divisions[0] <= old_end:
```

Comment:
I don't think these are identical, and the former is actually what we want. `fastparquet`'s `sorted_partitioned_columns` won't return stats if the column isn't properly partitioned, so the `None` check handles that case.

Reply:
(new error message looks good to me though)

Reply:
If `old_end is None` then the existing on-disk format doesn't have sorted divisions, but the new dataframe does. I don't think we should error in this case. If you agree, I should update the check in the arrow engine as well, I thought it already did this but it doesn't.

Reply:
> I don't think we should error in this case.

Hmm, I suppose I could be talked into that. In that case, let's update the arrow check

Reply:
Done. `fastparquet` always detects a column as unsorted (since it currently always loads metadata), while pyarrow only detects this case (now) if a `_metadata` file already exists. I think this is fine - this check is best effort, a false positive is easy to fix by passing in `ignore_divisions=True`.